internal/agent/device/lifecycle/manager.go

```diff
@@ -92,12 +102,85 @@ func (m *LifecycleManager) Initialize(ctx context.Context, status *v1alpha1.Devi
 }
 
 func (m *LifecycleManager) Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	// this controller currently does not implement a sync operation
 	return nil
 }
 
 func (m *LifecycleManager) AfterUpdate(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	if current.Decommission == nil && desired.Decommission != nil {
+		m.log.Warn("Detected decommissioning request from flightctl service")
+		if err := m.updateWithStartedCondition(ctx); err != nil {
+			return fmt.Errorf("failed to update decommission started status: %w", err)
+		}
+		m.log.Warn("Updated Condition to decommissioning started")
+
+		// TODO: add support for additional decommissioning target types.
+		// these are the steps that will take places between Started and Completed status
+
+		if err := m.updateWithCompletedCondition(ctx); err != nil {
+			return fmt.Errorf("failed to update decommission completed status: %w", err)
+		}
+		m.log.Warn("No errors during decommissioning, updated Condition to decommissioning completed")
+
+		// after this point the device will no longer be able to communicate with the management service
+		m.log.Warn("Preparing to wipe agent certificate and keys and reboot")
+		return m.wipeAndReboot(ctx)
```

Comment:
If this fails what should the agent do?

Reply:
I'm not sure there's much that it can do except retry, which may or may not be worth it. Thoughts?